docs/docs/behaviors/tap-dance.md

```diff
@@ -67,10 +69,41 @@ The following image describes the behavior of this particular tap-dance.
 ![Timing Diagram](../assets/tap-dance/timing_diagram.svg)
 
 :::note
-Alphanumeric [`key press`](key-press.md) bindings, like those used for `td0`,
-will release as soon as an interrupting key press occurs.
-For instance, if a modifier key like `LSHIFT` were to replace the `N1`
-binding in the last example above, it would remain pressed until `td0`'s
-binding is released and the output would instead be `J`. Any following
-alphanumeric key presses would be capitalized as long as `td0` is held down.
+Alphanumeric [`key press`](key-press.md) bindings, like those used for `td0`, will release as soon as an interrupting key press occurs. For instance, if a modifier key like `LSHIFT` were to replace the `N1` binding in the last example above, it would remain pressed until `td0`'s binding is released and the output would instead be `J`. Any following alphanumeric key presses would be capitalized as long as `td0` is held down.
 :::
+
+</TabItem>
+
+<TabItem value="advanced">
+
+This example configures a mod-tap inside a tap-dance named `td_mt` that outputs `LSHIFT` on a single tap, `CAPSLOCK` on a single press and hold, and `LCTRL` on double-tap.
```

Comment:
I believe that hold-tap (`&ht`) should be used here rather than mod-tap (`&mt`) because `CAPSLOCK` is not a modifier.

But I also think this example is a little confusing, because:

1. `SHIFT` is normally _held_, but here it is output on a _tap_
2. `CAPSLOCK` is normally _tapped_ to toggle on/off, but here it is output on a _hold_
3. `CONTROL` is also usually _held_, but here it is output on a double-_tap_

Apologies if I'm just misunderstanding, but I think the example might make more sense like so:

```suggestion
This example configures a hold-tap inside a tap-dance named `td_ht` that outputs `CAPSWORD` on a single tap, `SHIFT` on a single press and hold, and `CAPSLOCK` on double-tap.
```

This way, the tappy things are output with taps and the holdy things are output with holds.

Reply:
Couple of initial reactions:
- `CONTROL` still makes sense on the double-tap because the tap-dance doesn't release the `LCTRL` until its own binding is released
- `&ht` is not a pre-defined ZMK behavior; we'd have to create a separate hold-tap behavior that implements the `&caps_word` on single-tap and `&kp LSHIFT` on single-press-and-hold
  - Using `&mt` in the tap-dance achieves what most people are looking for in a tap-hold-dance, i.e., a different `&kp` for single-tap, single-press-and-hold, double-tap, double-press-and-hold, etc.

Otherwise, I agree with updating the example to ensure things that are usually held are output on holds. Perhaps we can update the existing example with the changes you've suggested and add another example that explicitly uses the `&mt` inside a tap-dance.

Reply:
Something like this for the updated version of the current example, perhaps: 

```suggestion
This example configures a hold-tap inside a tap-dance named `td_ht` that outputs `&caps_word` on a single tap, `&kp LSHIFT` on a single press and hold, and `&kp CAPSLOCK` on double-tap.

```title="Advanced Tap-Dance Example: Nested Hold-Tap"
#include <behaviors.dtsi>
#include <dt-bindings/zmk/keys.h>

/ {
	behaviors {
		ht: hold_tap{
            compatible = "zmk,behavior-hold-tap";
            label = "HOLD_TAP";
            #binding-cells = <2>;
            tapping-term-ms = <200>;
            flavor = "tap-preferred";
            bindings = <&kp>, <&caps_word>;
        };

		td_ht: tap_dance_hold_tap {
            compatible = "zmk,behavior-tap-dance";
            label = "TAP_DANCE_HOLD_TAP";
            #binding-cells = <0>;
            tapping-term-ms = <200>;
            bindings = <&ht LSHIFT 0>, <&kp CAPSLOCK>;
        };
	};

	keymap {
		compatible = "zmk,keymap";

		default_layer {
			bindings = <
	            &td_ht
			>;
		};
	};
};

```

Reply:
Oh very good points! 

I think a custom hold tap behavior overcomplicates things here. Also, because `&caps_word` doesn't take any parameters that makes for an awkward hold-tap example (i.e. `&ht LSHIFT 0`). I'm sorry for making such a poor suggestion. 😅 

I believe there is one issue with the original example though – `LSHIFT` is unusable as it cannot be held? If so, I think all we really need to do is swap `LSHIFT` and `CAPSLOCK` in the example?

```suggestion
This example configures a mod-tap inside a tap-dance named `td_mt` that outputs `CAPSLOCK` on a single tap, `LSHIFT` on a single press and hold, and `LCTRL` on double-tap.
```

Reply:
> Expanding on the third point, would it be worth mentioning in the mod-tap docs that the "hold" keycode doesn't necessarily need to be a modifier key?

Yep, I had never checked the code and always assumed it needed to be a modifier. I think a small tip mentioning this would be a nice touch.

Reply:
If simplicity is preferred, I'd be happy to just go in to change the original blurb for `td_mt` to swap the `CAPSLOCK` and `LSHIFT` keys, in addition to updating the mod-tap page to make note of the ability to use keycodes other than modifiers. Thought I'd also post the thing I've been working on that's a little more comprehensive, just in case:

### Updated Blurb
[Hold-taps](hold-tap.md) and tap-dances can be combined to create keybinds whose outputs are dependent on both the number and duration of the key presses. A simple example is shown below with two [mod-taps](mod-tap.md) nested inside a single tap-dance named `&mtd`, where the user can single-tap to output `&kp S`, press-and-hold once for `LSHIFT`, double-tap for `&kp A`, or double-tap and hold for `LALT`.

```title="Advanced Tap-Dance Example: Mod-Tap nested inside Tap-Dance"
#include <behaviors.dtsi>
#include <dt-bindings/zmk/keys.h>

/ {
	behaviors {
		mtd: mod_tap_dance {
			compatible = "zmk,behavior-tap-dance";
			label = "TAP_DANCE_HOLD_TAP";
			#binding-cells = <0>;
			tapping-term-ms = <200>;
			bindings = <&mt LSHIFT S>, <&mt LALT A>;
		};
	};

	keymap {
		compatible = "zmk,keymap";

		default_layer {
			bindings = <
				&mtd
			>;
		};
	};
};
```

The next example expands further on the idea of combining behaviors inside a single keybind. We define a custom [hold-tap](hold-tap.md) inside a tap-dance named `td_ht`, which outputs [Caps Word](caps-word.md) on a single tap, `LSHIFT` on a single press-and-hold, and `CAPSLOCK` when the tap-dance is pressed twice.

```title="Advanced Tap-Dance Example: Custom Hold-Tap nested inside Tap-Dance"
#include <behaviors.dtsi>
#include <dt-bindings/zmk/keys.h>

/ {
	behaviors {
		ht: hold_tap{
			compatible = "zmk,behavior-hold-tap";
			label = "HOLD_TAP";
			#binding-cells = <2>;
			tapping-term-ms = <200>;
			flavor = "tap-preferred";
			bindings = <&kp>, <&caps_word>;
		};

		td_ht: tap_dance_hold_tap {
			compatible = "zmk,behavior-tap-dance";
			label = "TAP_DANCE_HOLD_TAP";
			#binding-cells = <0>;
			tapping-term-ms = <200>;
			bindings = <&ht LSHIFT 0>, <&kp CAPSLOCK>;
		};
	};

	keymap {
		compatible = "zmk,keymap";

		default_layer {
			bindings = <
				&td_ht
			>;
		};
	};
};
```

I think the docs need a bit more indication somewhere, or again for the sake of redundancy, wherever it's relevant, that behaviors can be combined. At least with this updated blurb, there's a bit of progression in the "difficulty" of implementing the nested behaviors.

Reply:
In the meantime, however, I'll just commit the swapping of `CAPSLOCK` and `LSHIFT`. It's a pretty simple fix. We can decide to make the bigger change after.

Reply:
> Yep, I had never checked the code and always assumed from our documentation that it needed to be a modifier. I think a small tip mentioning this would be a nice touch.

![image](https://user-images.githubusercontent.com/27038994/172293174-6db85076-7910-4aec-a9eb-5bc8d48a0800.png)

Reply:
> 



> ### Reply
> If simplicity is preferred, I'd be happy to just go in to change the original blurb for `td_mt` to swap the `CAPSLOCK` and `LSHIFT` keys, in addition to updating the mod-tap page to make note of the ability to use keycodes other than modifiers. Thought I'd also post the thing I've been working on that's a little more comprehensive, just in case:
> 
> ### Updated Blurb
> [Hold-taps](hold-tap.md) and tap-dances can be combined to create keybinds whose outputs are dependent on both the number and duration of the key presses. A simple example is shown below with two [mod-taps](mod-tap.md) nested inside a single tap-dance named `&mtd`, where the user can single-tap to output `&kp S`, press-and-hold once for `LSHIFT`, double-tap for `&kp A`, or double-tap and hold for `LALT`.
> 
> ```
> #include <behaviors.dtsi>
> #include <dt-bindings/zmk/keys.h>
> 
> / {
> 	behaviors {
> 		mtd: mod_tap_dance {
> 			compatible = "zmk,behavior-tap-dance";
> 			label = "TAP_DANCE_HOLD_TAP";
> 			#binding-cells = <0>;
> 			tapping-term-ms = <200>;
> 			bindings = <&mt LSHIFT S>, <&mt LALT A>;
> 		};
> 	};
> 
> 	keymap {
> 		compatible = "zmk,keymap";
> 
> 		default_layer {
> 			bindings = <
> 				&mtd
> 			>;
> 		};
> 	};
> };
> ```
> 
> The next example expands further on the idea of combining behaviors inside a single keybind. We define a custom [hold-tap](hold-tap.md) inside a tap-dance named `td_ht`, which outputs [Caps Word](caps-word.md) on a single tap, `LSHIFT` on a single press-and-hold, and `CAPSLOCK` when the tap-dance is pressed twice.
> 
> ```
> #include <behaviors.dtsi>
> #include <dt-bindings/zmk/keys.h>
> 
> / {
> 	behaviors {
> 		ht: hold_tap{
> 			compatible = "zmk,behavior-hold-tap";
> 			label = "HOLD_TAP";
> 			#binding-cells = <2>;
> 			tapping-term-ms = <200>;
> 			flavor = "tap-preferred";
> 			bindings = <&kp>, <&caps_word>;
> 		};
> 
> 		td_ht: tap_dance_hold_tap {
> 			compatible = "zmk,behavior-tap-dance";
> 			label = "TAP_DANCE_HOLD_TAP";
> 			#binding-cells = <0>;
> 			tapping-term-ms = <200>;
> 			bindings = <&ht LSHIFT 0>, <&kp CAPSLOCK>;
> 		};
> 	};
> 
> 	keymap {
> 		compatible = "zmk,keymap";
> 
> 		default_layer {
> 			bindings = <
> 				&td_ht
> 			>;
> 		};
> 	};
> };
> ```
> 
> ### Reply continued
> I think the docs need a bit more indication somewhere, or again for the sake of redundancy, wherever it's relevant, that behaviors can be combined. At least with this updated blurb, there's a bit of progression in the "difficulty" of implementing the nested behaviors.

Here's a comparison of what the simple vs. more comprehensive updates to this section look like:

### Simple fix to original blurb - Quick implementation of mod-taps inside tap-dances

![image](https://user-images.githubusercontent.com/27038994/172293449-22b751eb-555e-4470-93dc-7f7b89cb51a6.png)

### Updated Blurb - More detail into combining behaviors

![image](https://user-images.githubusercontent.com/27038994/172293622-1bfdb794-58d0-4078-ad6f-120678d64fc1.png)

![image](https://user-images.githubusercontent.com/27038994/172293580-b79581ca-f552-40b9-8890-0ffd728d8d81.png)

Reply:
If we go the more comprehensive route, a note somewhere about the purpose of the dummy `0` in the `&ht LSHIFT 0` would probably help too.

Reply:
I think if we remove the "Additional information" header then the info box would look OK, what do you think?

Reply:
Works for me. Any comments on the wording for this section of the mod-tap docs? Will repost here just for readability:

> Under the hood, the mod-tap is simply a ["hold-preferred" flavor](hold-tap.md/#flavors) of [hold-tap](hold-tap.md) with a [`tapping-term-ms`](hold-tap.md/#tapping-term-ms) of 200 that takes in two [keypresses](key-press.md) as its "hold" and "tap" parameters. This means that the mod-tap can be used to invoke **any** [keycode](../codes/index.mdx), and is _not_ limited to only activating [modifier keys](../codes/modifiers.mdx) when it is held.

Reply:
I think the wording there is pretty good -- just a heads up that the link for "keycode" in your quote is 404'ing.

We are emphasizing that these are for key presses (via the keycode link), but is it worth also emphasizing it doesn't work for other types of bindings (in which case we can refer to layer-tap and hold-tap docs)? I couldn't be sure but I thought I'd throw it out there.

Reply:
Good eye, but not to worry; on local testing, the link to `keycode` works fine since it's a relative path. As for mentioning that mod-taps can't take other types of bindings, it's another valid suggestion. I'll take a stab at writing a draft up right now.

Reply:
> If we go the more comprehensive route, a note somewhere about the purpose of the dummy `0` in the `&ht LSHIFT 0` would probably help too.

![image](https://user-images.githubusercontent.com/27038994/172314702-9b142284-4d36-4acb-a649-d5aa9b1caae7.png)

Reply:
I'm thinking we expand this tip section I'm working on to mention that the dummy value technique applies to all nested, zero-binding-cell behaviors too. For now, I don't have the brainpower to get to it at the moment, but I can take another shot at it tomorrow.

Reply:
> I think the docs need a bit more indication somewhere, or again for the sake of redundancy, wherever it's relevant, that behaviors can be combined. 

@kurtis-lew I agree. I think maybe we could explore this goal separately though? Maybe we need a separate GitHub issue dedicated to documentation about explaining behaviors and using them in more advanced ways. (Personally I think it would be good to keep the pages for individual behaviours focused on their own functionality and usage.)

The blurb you've got in the PR currently is great, and I think for now it would be good not to expand the scope of this PR too much, so that we can merge the other good work that's already done.

Reply:
Good point. I think I'll just come back to the code snippets posted here to copy-paste into another PR related to composite behaviors if need be. I think if we get the go-ahead from @okke-formsma, we should be good to merge!